runners/spark/src/main/java/org/apache/beam/runners/spark/io/MicrobatchSource.java

```diff
@@ -201,56 +201,66 @@ public int hashCode() {
    */
```

Comment:
This looks like either a rewrite or a near-clone of `BoundedReadFromUnboundedSource` -- any opportunity for code reuse?